lake_hylia.logic

```diff
@@ -1,3 +1,93 @@
 area LakeHylia:
-  room BeanstalkArea
-  room Main
+  room BeanstalkArea:
+    node EastArea:
+      door LeftLadder
+      door RightLadder
+    node WestArea:
+      door Ladder
+    node BeanstalkFusion:
+      entrance beanStalkFusion
+    node BeanStalkFusionExit:
+      exit beanStalkFusion
+    BeanStalkFusionExit -> WestArea: flag HyliaBeanstalkFusion
+    WestArea -> BeanstalkFusion: flag HyliaBeanstalkFusion
+    node EastArea +: exit Virtual.Zeffa.FluteAway
+    node WestArea +: exit Virtual.Zeffa.FluteAway
+  room Main:
+    node Crest
+      entrance Crest
+      door MinishHouse
+      exit CrestHoleFusion
+      flag Virtual.Zeffa.LakeCrest
+    Crest -> NorthWest
+    node CrestHoleFusion:
+      entrance CrestHoleFusion
+    Crest -> CrestHoleFusion: flag CrestHoleFusion
+    node MinishWater:
+      exit MinishSouth
+      exit NorthHole
+    node MinishWaterEntrances:
+      entrance South
+      entrance NorthHole
+    MinishWater -> MinishWaterEntrances: item Flippers
+    node EastVineHole:
+      door EastVineHole
+    MinishWater <-> EastVineHole: item Flippers
+    node MinishPortal:
+      door MinishPath
+    MinishPortal <-> MinishWater: item Flippers
+    MinishPortal -> SouthEast: item CanGrow
+    node MinishPortalBonk:
+      flag TreeBonk
+    SouthEast -> MinishPortalBonk: item Boots
+    SouthEast -> MinishPortal: item CanTurnMinish
+    node DigCaveFusion:
+      door DigCave
+    Water <-> DigCaveFusion: flag HyliaDigCaveFusion
+    node IslandTop:
+      door MinishHouse
+      door Hole
+    IslandTop -> MinishWater: item Flippers
+    node SouthWest:
+      door East
+      door EastSouth
+      door South
+      enemy SouthWestChuChus
+    node Water:
+      chest SouthHP
+      enemy Octoroks
+    node DropletsEntrance:
+      door DropletsEntrance
+    Water <-> DropletsEntrance: (item Cape | item Flippers)
+    node FifiWaterHole:
+      item HP
+    NorthWest <-> FifiWaterHole: item Flippers
+    node West:
+      door Tree
+      enemy Octoroks
+    West <-> Water: (item Cape | item Flippers)
+    West <-> NorthWest: flag HyliaBoulder
+    node NorthWest:
+      door Fifi
+      door West
+    node SouthEast:
+      door Ladder
+      door Cabin
+    SouthEast <-> Water: (item Cape | item Flippers)
+    node DigCaveNorthArea:
+      exit DigCave
+    node NorthHP:
+      item NorthHP
+    DigCaveNorthArea <-> NorthHP: item Cape
+    node DigCaveNorthEntrance:
+      entrance DigCave
+    DigCaveNorthArea -> DigCaveNorthEntrance: item MoleMitts
+    DigCaveNorthArea <-> NorthWest: item Cape
+    NorthWest <-> Water: (item Cape | item Flippers)
+    node Crest +: exit Virtual.Zeffa.FluteAway
+    node DigCaveFusion +: exit Virtual.Zeffa.FluteAway
+    node SouthEast +: exit Virtual.Zeffa.FluteAway
+    node Water +: exit Virtual.Zeffa.FluteAway
+    node DigCaveNorthArea +: exit Virtual.Zeffa.FluteAway
+    node West +: exit Virtual.Zeffa.FluteAway
+    node SouthWest +: exit Virtual.Zeffa.FluteAway
```

Comment:
Keeping this to a minimal set: NorthWest is missing. You can always reach NorthWest from Crest. You cannot reach water without also reaching another node that has the entrance. DigCaveFusion needs to be redone with the split to that node - you can only flute away from there if the fusion is done.
And, as mentioned with the `Crest` exit, all of these should be `entrances` - something you go somewhere else with.
```suggestion
    node NorthWest +: entrance Virtual.Zeffa.FluteAway
    node DigCaveNorthArea +: entrance Virtual.Zeffa.FluteAway
    node SouthEast +: entrance Virtual.Zeffa.FluteAway
    node SouthWest +: entrance Virtual.Zeffa.FluteAway
    node DigCaveFusion +: entrance Virtual.Zeffa.FluteAway
```